### PR TITLE
Randomize some settings in functional tests

### LINKED ISSUE #32268: Randomization of settings in functional tests.
We will create a list of settings and distributions of their values.
The settings will be randomly set before running functional tests, according to the distributions.
This can be implemented as a Python code directly inside `clickhouse-test` (no need for any configs or DSLs).

Examples:
- enable `convert_query_to_cnf` with probability 1/2;
- set `max_block_size` uniformly random in a range of 8000..100000;
- set `max_threads` uniformly random in a range of 1..64;

If a test needs fixed value of some setting, it should set it explicitly in the test.

### PR DESCRIPTION
**Pull request by Avogar**
Changelog category (leave one):

- Build/Testing/Packaging Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Randomize some settings in functional tests. This closes #32268.

### CODE REVIEW

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
optimize_move_to_prewhere = 0|1

**Comment by alexey-milovidov** [comment]
prefer_localhost_replica = 0|1

**Comment by alexey-milovidov** [comment]
group_by_two_level_threshold
group_by_two_level_threshold_bytes

With 10% probability set to very low value (to activate always) and with 10% set to very high value (to never activate)

**Comment by alexey-milovidov** [comment]
distributed_aggregation_memory_efficient = 0|1

**Comment by alexey-milovidov** [comment]
allow_experimental_parallel_reading_from_replicas = 0|1

**Comment by alexey-milovidov** [comment]
max_insert_threads with probability 1/2 set it uniformly random 1..16

**Comment by alexey-milovidov** [comment]
output_format_parallel_formatting = 0|1

**Comment by alexey-milovidov** [comment]
PS. We can adapt existing tests then add more randomization in subsequent PR.

**Comment by alexey-milovidov** [comment]
@den-crane @SaltTan Maybe you can recall some problematic settings from your experience.

File: tests/clickhouse-test [hunk 1]
```diff
@@ -336,6 +336,24 @@ class FailureReason(enum.Enum):
     INTERNAL_ERROR = "Test internal error: "
 
 
+class SettingsRandomizer:
+    settings = {
+        "convert_query_to_cnf": lambda: random.randint(0, 1),
+        "max_block_size": lambda: random.randint(8000, 100000),
+        "max_threads": lambda: random.randint(1, 64),
+        "input_format_parallel_parsing": lambda: random.randint(0, 1),
```
Comment (tavplubix) [inline tests/clickhouse-test:344]:
```suggestion
        "input_format_parallel_parsing": lambda: random.randint(0, 1),
        "min_chunk_bytes_for_parallel_parsing": lambda: max(1024, int(random.gauss(10 * 1024 * 1024, 5 * 1000 * 1000))),
        "output_format_parallel_formatting": lambda: random.randint(0, 1),
        "fsync_metadata": lambda: random.randint(0, 1),
        "priority": lambda: int(abs(random.gauss(0, 2))),
```

**Comment by tavplubix** [comment]
We should print changed settings somewhere in logs to simplify debugging

File: tests/clickhouse-test [hunk 2]
```diff
@@ -668,6 +690,7 @@ class TestCase:
 
             self.runs_count += 1
             self.testcase_args = self.configure_testcase_args(args, self.case_file, suite.suite_tmp_path)
+            client_options = SettingsRandomizer.add_random_settings(client_options)
```
Comment (azat) [inline tests/clickhouse-test:693]: Maybe also worth applying them to `client_options_query_str`/`env['CLICKHOUSE_URL_PARAMS']` ?

**Comment by Algunenano** [comment]
Love the idea but do you really want to introduce them all at once? The last similar thing done was to run under a random timezone and it took weeks and many PRs to stabilize things.

**Comment by alexey-milovidov** [comment]
I'm ok if it will take weeks.

**Comment by Avogar** [comment]
@Mergifyio update

**Comment by SaltTan** [comment]
> @den-crane @SaltTan Maybe you can recall some problematic settings from your experience.

I'm sorry it took me so long.
- optimize*
- enable*
- use*
- compile_aggregate_expressions
- compile_expressions
- There was an issue with 'read_backoff' once.

**Comment by Avogar** [comment]
@Mergifyio update

**Pull request merged**